xtable-api/src/main/java/org/apache/xtable/conversion/TargetTable.java

```diff
@@ -44,4 +44,8 @@ public TargetTable(
     this.metadataRetention =
         metadataRetention == null ? Duration.of(7, ChronoUnit.DAYS) : metadataRetention;
   }
+
+  public String getId() {
```

Comment:
What is the contract of Id? Could you please add javadoc for public methods of the core classes ?

Reply:
I have removed this to avoid confusion and using `Map<TargetTable, List<TargetCatalogConfig>> targetCatalogs` instead.